exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/Otel2PrometheusConverterTest.java

```diff
@@ -0,0 +1,257 @@
+            "sample_ratio gauge",
+            "sample_ratio description",
+            "sample_ratio"),
+        // Gauge without "1" as unit
+        // Previously metricName was "sample_unit"
+        Arguments.of(
+            createSampleMetricData("sample", "unit", MetricDataType.LONG_GAUGE),
+            "sample_unit gauge",
+            "sample_unit description",
+            "sample_unit"),
+        // special case with counter
+        // Previously metricName was "sample_unit_total"
+        Arguments.of(
+            createSampleMetricData("sample", "unit", MetricDataType.LONG_SUM),
+            "sample_unit_total counter",
+            "sample_unit_total description",
+            "sample_unit_total"),
+        // special case unit "1", but not gauge - "1" is dropped
+        // Previously metricName was "sample_total"
+        Arguments.of(
+            createSampleMetricData("sample", "1", MetricDataType.LONG_SUM),
+            "sample_ratio_total counter",
+            "sample_ratio_total description",
+            "sample_ratio_total"),
+        // units expressed as numbers other than 1 are retained
+        // Previously metricName was "sample_2_total"
+        Arguments.of(
+            createSampleMetricData("sample", "2", MetricDataType.LONG_SUM),
+            "sample_2_total counter",
+            "sample_2_total description",
+            "sample_2_total"),
+        // metric name with unsupported characters
+        // Previously metricName was "s_ple_percent_per_minute"
+        Arguments.of(
+            createSampleMetricData("s%%ple", "%/min", MetricDataType.SUMMARY),
+            "s__ple_percent_per_minute summary",
+            "s__ple_percent_per_minute description",
+            "s__ple_percent_per_minute_count"),
+        // metric name with dropped portions
+        // Previously metricName was "s_ple_percent_per_minute"
+        Arguments.of(
+            createSampleMetricData("s%%ple", "%/min", MetricDataType.SUMMARY),
+            "s__ple_percent_per_minute summary",
+            "s__ple_percent_per_minute description",
+            "s__ple_percent_per_minute_count"),
+        // metric unit as a number other than 1 is not treated specially
+        // Previously metricName was "metric_name_2"
+        Arguments.of(
+            createSampleMetricData("metric_name", "2", MetricDataType.SUMMARY),
+            "metric_name_2 summary",
+            "metric_name_2 description",
+            "metric_name_2_count"),
+        // metric unit is not appended if the name already contains the unit
+        // Previously metricName was "metric_name_total"
+        Arguments.of(
+            createSampleMetricData("metric_name_total", "total", MetricDataType.LONG_SUM),
+            "metric_name_total counter",
+            "metric_name_total description",
+            "metric_name_total"),
+        // metric unit is not appended if the name already contains the unit - special case for
```

Comment:
The comment here also needs to be updated - explaining this case around reserved words in prometheus IIUC.